### PR TITLE
Update basic erc20/ether HTLC RFC

### LINKED ISSUE #135: Update Ether/Erc20 HTLC
# Problem

Our current HTLCs did not fail nor provide useful information in a failure case (such as redeem with wrong secret or refund too early).
With this PR we solved this problem: https://github.com/comit-network/blockchain-contracts/pull/37/
If we start using the new blockchain_contracts we will have RFCs mismatches.

# Goal
Have no mismatches between code and RFC ;)

# Latest Contract Offsets: 

## ** Ether on Ethereum **
Contract template:
 `61012861000f6000396101286000f3361561007957602036141561004f57602060006000376020602160206000600060026048f17f100000000000000000000000000000000000000000000000000000000000000160215114166100ae575b7f696e76616c69645365637265740000000000000000000000000000000000000060005260206000fd5b426320000002106100eb577f746f6f4561726c7900000000000000000000000000000000000000000000000060005260206000fd5b7f72656465656d656400000000000000000000000000000000000000000000000060206000a1733000000000000000000000000000000000000003ff5b7f726566756e64656400000000000000000000000000000000000000000000000060006000a1734000000000000000000000000000000000000004ff`
| Name | Byte Range | Length (bytes) |
|:--- |:--- |:--- |
| `secret_hash` | 53..85 | 32 |
| `refund_timestamp` | 139..143 | 4 |
| `redeem_address` | 229..249 | 20 |
| `refund_address` | 290..310 | 20 |

## ** ERC20 on Ethereum **
Contract template:
 `61018c61000f60003961018c6000f3361561007957602036141561004f57602060006000376020602160206000600060026048f17f100000000000000000000000000000000000000000000000000000000000000160215114166100ae575b7f696e76616c69645365637265740000000000000000000000000000000000000060005260206000fd5b426320000002106100f1577f746f6f4561726c7900000000000000000000000000000000000000000000000060005260206000fd5b7f72656465656d656400000000000000000000000000000000000000000000000060206000a1733000000000000000000000000000000000000003602052610134565b7f726566756e64656400000000000000000000000000000000000000000000000060006000a1734000000000000000000000000000000000000004602052610134565b63a9059cbb6000527f5000000000000000000000000000000000000000000000000000000000000005604052602060606044601c6000736000000000000000000000000000000000000006620186a05a03f150602051ff`
| Name | Byte Range | Length (bytes) |
|:--- |:--- |:--- |
| `secret_hash` | 53..85 | 32 |
| `refund_timestamp` | 139..143 | 4 |
| `redeem_address` | 229..249 | 20 |
| `refund_address` | 296..316 | 20 |
| `amount` | 333..365 | 32 |
| `token_contract_address` | 379..399 | 20 |

### PR DESCRIPTION
**Pull request by bonomat**
Resolves #135

### CODE REVIEW

File: RFC-007-SWAP-Basic-Ether.adoc [hunk 1]
```diff
@@ -207,18 +214,33 @@ The contract emits two logs:
...
-| `Refunded()`
-| 0x5d26862916391bf49478b2f5103b0720a842b45ef145a268f2cd1fb2aed55178
+| ascii to hex of `refunded`
+| 0x726566756e646564000000000000000000000000000000000000000000000000
 | -
 | The contract is refunded
 
 |===
 
+The contract returns two different error values the failure case:
```
Comment (da-kami) [inline RFC-007-SWAP-Basic-Ether.adoc]:
```suggestion
The contract returns two different error values for the failure cases:
```

File: RFC-007-SWAP-Basic-Ether.adoc [hunk 2]
```diff
@@ -272,6 +294,13 @@ To refund the HTLC, the funder MUST send a transaction to `contract_address` wit
 
 To be notified of the refund event, both parties SHOULD watch the blockchain for  `contract_address` emitting the `Refunded()` log.
 
+=== Failure
+
+A transaction to the HTLC will fail in two cases:
+
+. *If redeem was attempted with an invalid secret*: the contract will return `invalidSecret` int bytes (`0x696e76616c69645365637265740000000000000000000000000000000000000000`)
```
Comment (da-kami) [inline RFC-007-SWAP-Basic-Ether.adoc]:
```suggestion
. *If redeem was attempted with an invalid secret*: the contract will return `invalidSecret` in bytes (`0x696e76616c69645365637265740000000000000000000000000000000000000000`)
```

File: RFC-007-SWAP-Basic-Ether.adoc [hunk 3]
```diff
@@ -272,6 +294,13 @@ To refund the HTLC, the funder MUST send a transaction to `contract_address` wit
...
+. *If refund was attempted too early*: the contract will return `tooEarly` int bytes (`0x746f6f4561726c79000000000000000000000000000000000000000000000000`)
```
Comment (da-kami) [inline RFC-007-SWAP-Basic-Ether.adoc]:
```suggestion
. *If refund was attempted too early*: the contract will return `tooEarly` in bytes (`0x746f6f4561726c79000000000000000000000000000000000000000000000000`)
```

File: RFC-009-SWAP-Basic-ERC20.adoc [hunk 4]
```diff
@@ -136,18 +139,40 @@ check_expiry:
...
+    32 hash return value
+    ->
+    0 transfer pointer
+    32 to
+    64 amount
+    96 transfer return
+
+    place holders
+    0x3000000000000000000000000000000000000003 // redeem address
+    0x4000000000000000000000000000000000000004 // refund address
```
Comment (da-kami) [inline RFC-009-SWAP-Basic-ERC20.adoc]: Where is ` 0x5000000000000000000000000000000000000005`? Is this a typo or is it actually 03, 04, 06, 07?

File: RFC-009-SWAP-Basic-ERC20.adoc [hunk 5]
```diff
@@ -189,39 +215,55 @@ To compile the contract code replace the placeholder values with the HTLC paramt
...
-| `Refunded()`
-| 0x5d26862916391bf49478b2f5103b0720a842b45ef145a268f2cd1fb2aed55178
+| ascii to hex of `refunded`
+| 0x726566756e646564000000000000000000000000000000000000000000000000
 | -
 | The contract is refunded
+
+|===
+
+The contract returns two different error values the failure case:
```
Comment (da-kami) [inline RFC-009-SWAP-Basic-ERC20.adoc]:
```suggestion
The contract returns two different error values for the failure cases:
```

File: RFC-009-SWAP-Basic-ERC20.adoc [hunk 6]
```diff
@@ -295,6 +337,13 @@ To refund the HTLC, the funder MUST send a transaction to `htlc_contract` with e
 
 To be notified of the refund event, both parties SHOULD watch the blockchain for `htlc_contract` emitting the `Refunded()` log.
 
+=== Failure
+
+A transaction to the HTLC will fail in two cases:
+
+. *If redeem was attempted with an invalid secret*: the contract will return `invalidSecret` int bytes (`0x696e76616c69645365637265740000000000000000000000000000000000000000`)
```
Comment (da-kami) [inline RFC-009-SWAP-Basic-ERC20.adoc]:
```suggestion
. *If redeem was attempted with an invalid secret*: the contract will return `invalidSecret` in bytes (`0x696e76616c69645365637265740000000000000000000000000000000000000000`)
```

File: RFC-009-SWAP-Basic-ERC20.adoc [hunk 7]
```diff
@@ -295,6 +337,13 @@ To refund the HTLC, the funder MUST send a transaction to `htlc_contract` with e
...
+. *If refund was attempted too early*: the contract will return `tooEarly` int bytes (`0x746f6f4561726c79000000000000000000000000000000000000000000000000`)
```
Comment (da-kami) [inline RFC-009-SWAP-Basic-ERC20.adoc]:
```suggestion
. *If refund was attempted too early*: the contract will return `tooEarly` in bytes (`0x746f6f4561726c79000000000000000000000000000000000000000000000000`)
```

File: RFC-009-SWAP-Basic-ERC20.adoc (hunk 4 above)
Comment (bonomat) [reply RFC-009-SWAP-Basic-ERC20.adoc]: no bug, feature (0x5... is the placeholder for the amount) :)
However, let me remove this from the RFC. no need to have this comment in here.

**Pull request merged**